firebase-appdistribution/src/main/java/com/google/firebase/appdistribution/impl/FeedbackSender.java

```diff
@@ -54,6 +72,85 @@ private Task<String> attachScreenshot(String feedbackName, @Nullable Uri screens
     if (screenshotUri == null) {
       return Tasks.forResult(feedbackName);
     }
-    return testerApiClient.attachScreenshot(feedbackName, screenshotUri);
+    if (!screenshotUri.getScheme().equals("file") && !screenshotUri.getScheme().equals("content")) {
+      return Tasks.forException(
+          new FirebaseAppDistributionException(
+              String.format(
+                  "Screenshot URI '%s' was not a content or file URI. Not starting feedback.",
+                  screenshotUri),
+              Status.UNKNOWN));
+    }
+    return testerApiClient.attachScreenshot(
+        feedbackName,
+        screenshotUri,
+        getScreenshotFilename(screenshotUri),
+        getContentType(screenshotUri));
+  }
+
+  private String getContentType(Uri screenshotUri) {
+    String contentType = null;
+    if (screenshotUri.getScheme().equals("file")) {
+      contentType = getContentTypeForFilename(screenshotUri.getLastPathSegment());
+    } else if (screenshotUri.getScheme().equals("content")) {
+      contentType = applicationContext.getContentResolver().getType(screenshotUri);
+    }
+    if (contentType == null) {
+      LogWrapper.w(
+          TAG,
+          String.format("Could not get content type for URI %s. Assuming PNG.", screenshotUri));
+      return CONTENT_TYPE_PNG;
+    }
+    return contentType;
+  }
+
+  private String getScreenshotFilename(Uri screenshotUri) {
+    if (screenshotUri.getScheme().equals("file")) {
+      return screenshotUri.getLastPathSegment();
+    } else {
+      return getContentFilename(screenshotUri);
+    }
+  }
+
+  private String getContentFilename(Uri contentUri) {
+    ContentResolver contentResolver = applicationContext.getContentResolver();
+    try (Cursor returnCursor =
+        contentResolver.query(
+            contentUri,
+            new String[] {OpenableColumns.DISPLAY_NAME},
+            /* selection= */ null,
+            /* projectionArgs= */ null,
+            /* sortOrder= */ null)) {
+      if (returnCursor == null) {
+        LogWrapper.w(
+            TAG,
+            String.format(
+                "Unable to get filename from URI '%s', using default '%s'",
+                contentUri, DEFAULT_FILENAME));
+        return DEFAULT_FILENAME;
```

Comment:
This should probably be based on the content type (if we were able to figure that out), so have a default filename for each (supported) content type.